src/DateInput/DateInput.tsx

```diff
@@ -192,6 +192,7 @@ export default class DateInput extends React.Component<Props, State> {
             selected={this.props.value}
             popperClassName="DatePicker--Popper"
             popperPlacement={this.props.popperPlacement}
+            onChange={this.props.onChange}
```

Comment:
That seems like a good idea. My goal here was to avoid any potential breaking behavior, but I think my change is technically breaking since `onChange` would be called where it isn't currently been called.

Reply:
Something I thought of while typing `DateInput`: should we get rid of the option rendering of two different sub components? It makes it harder to type things because the subcomponents share some props, but not all and they have some props with the same name but different type. I think it's also kind of weird because these are two hefty components, and it's not like conditionally rendering a `div` or a `span`. 

When I checked the usage of `DateInput` in our codebase, I found only one place where `useTime` was used, and it's in the portal. Maybe we can simplify this component and add a shared component in the portal.

Reply:
I definitely agree with removing the conditional rendering of two different underlying components. But I also think that we should maintain a `useTime` option in clever-components. It feels like a generally useful option. I wonder if `<ReactDateTime>` supports rendering without a time picker 🤔 That said, these changes are probably out of scope for this PR.

-----

Going back to the original question of `onSelect` vs. `onChange`, it seems like supplying both an `onSelect` and an `onChange` will result in redundant updates. Since `onChange` should take care of all needed updates, I imagine the `onSelect` --> `onChange` transition should be safe

Reply:
Ok, that's fine with me. Another option to consider is to expose a new `onSelect` prop, but we can always add that later if we need it.

Reply:
`onSelect` has been eliminated 🔪